packages/smooth_app/lib/pages/home_page.dart

```diff
@@ -80,279 +80,36 @@ class _HomePageState extends State<HomePage> {
   }
 }
 
-// Note: In addition to the history of scanned products, this page currently
-// contains the user's lists and other stuff from the old Smoothie home page.
-// The intention is to move those features to other screens later or remove them
-// altogether.
-class OldHomePage extends StatefulWidget {
-  const OldHomePage();
-
+class HistoryPage extends StatefulWidget {
+  const HistoryPage();
   @override
-  State<OldHomePage> createState() => _OldHomePageState();
+  State<HistoryPage> createState() => _HistoryPageState();
 }
 
-class _OldHomePageState extends State<OldHomePage> {
-  static const ColorDestination _COLOR_DESTINATION_FOR_ICON =
-      ColorDestination.SURFACE_FOREGROUND;
-  static const Icon _ICON_ARROW_FORWARD = Icon(Icons.arrow_forward);
-
-  late DaoProductList _daoProductList;
+class _HistoryPageState extends State<HistoryPage> {
+  late DaoProductList _daoProductList ;
+  final productList = ProductList(
+    listType: ProductList.LIST_TYPE_HISTORY,
+    parameters: '',
+  );
+  var loadedList = false;
 
   @override
   Widget build(BuildContext context) {
     final LocalDatabase localDatabase = context.watch<LocalDatabase>();
     _daoProductList = DaoProductList(localDatabase);
-    final ThemeData themeData = Theme.of(context);
-    final ColorScheme colorScheme = themeData.colorScheme;
+    if (!loadedList) {
```

Comment:
Will the ProductList always be in sync after the first load, so for example if we scan another product after this list is loaded, will the new product be automatically added to the list?

Reply:
Not currently, no - we will need a way to update `productList` when we need it to.

For example, we could have a global timestamp for the last viewed product, and use that instead of `loadedList` to check if we need to reload the list from the dao. WDYT?

Reply:
so you mean there would be 2 timesstamps, one in this class, another in productlist (I see that there are multiple types of product list so each type would need a timestamp?) and then we fetch only if the last_fetched_timestamp in this class < last_viewed_timestamp in the Dao ?

Something like that? Sorry if I misunderstood.

Reply:
I'm guessing most types of product lists are unique and need to be reloaded on widget creation but never on widget build - but this one (list of products we have looked at) is global and needs to rebuild the widget if you look at a new product, right? What I'm trying to get at is that this particular list is maybe different from the others.

Thus, I think a single pair of timestamps (last loaded, last changed) for that particular product list type would be fine.

Another (perhaps better, more canonical Flutter) alternative would be a Provider or ValueNotifier that contains an object only used for resetting this list, and make the build function subscribe to changes in it via `context.watch` or `ValueListenableBuilder`. I actually like that more - it feels more in line with the rest of the app.

Even better, of course, would be a database that allowed us to subscribe to updates to particular queries (I'm looking at you, Cloud Firestore ;), but we don't have that.

Reply:
Working on an example using Provider, but not yet done...

Reply:
Alright, I couldn't stay away. Look now - `ProductPage.build` calls `context.read<HistoryUpdatedSemaphore>().notify()`, which will make the `HistoryPage` rebuild (if it's still around), otherwise just reload history on next build.

Reply:
Using `didChangeDependencies` instead as discussed.